app/src/main/javascript/bundles/experiment-page/src/ExperimentPageRouter.js

```diff
@@ -24,11 +24,30 @@ const TabCommonPropTypes = {
 }
 
 // What component each tab type should render, coupled to ExperimentController.java
-const tabTypeComponent = {
-  'results' : TSnePlotViewRoute,
-  'experiment-design' : ExperimentDesignRoute,
-  'supplementary-information' : SupplementaryInformationRoute,
-  'downloads' : DownloadsRoute
+let tabTypeComponent = []
+
+function enableExperimentPageTab(tab) {
+  if (tab.type === 'results') {
+    if (Array.isArray(tab.props.ks) && (tab.props.plotTypesAndOptions)) {
```

Comment:
This can be cleaned a bit more:
- it is repetitive
- part of the validation can be extracted to a well named function to make it clean what it is doing and hide the details